frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+		pub MappingFactoryAddress get(fn mapping_factory_address) config(): EthereumAddress;
+		pub EthereumBackingAddress get(fn ethereum_backing_address) config(): EthereumAddress;
+		pub VerifiedIssuingProof
+			get(fn verified_issuing_proof)
+			: map hasher(blake2_128_concat) EthereumTransactionIndex => bool = false;
+		pub BurnTokenEvents get(fn burn_token_events): Vec<<T as frame_system::Config>::Event>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Config> for enum Call
+	where
+		origin: T::Origin
+	{
+		const FeeEstimate: RingBalance<T> = T::FeeEstimate::get();
+
+		fn deposit_event() = default;
+
+		fn on_initialize(_n: BlockNumber<T>) -> Weight {
+			<BurnTokenEvents<T>>::kill();
+			0
+		}
+
+		#[weight = <T as darwinia_evm::Config>::GasWeightMapping::gas_to_weight(0x100000)]
+		pub fn register_or_redeem_erc20(origin, proof: EthereumReceiptProofThing<T>) {
+			log::trace!(target: "darwinia-issuing", "start to register_or_issuing_erc20");
+			let user = ensure_signed(origin)?;
+			let tx_index = T::EthereumRelay::gen_receipt_index(&proof);
+			ensure!(!VerifiedIssuingProof::contains_key(tx_index), <Error<T>>::AssetAR);
+			let verified_receipt = T::EthereumRelay::verify_receipt(&proof)
+				.map_err(|err| {
+					log::trace!(target: "darwinia-issuing", "verify error {:?}", err);
+					<Error<T>>::ReceiptProofInv
+				})?;
+
+			let register_event = Abi::register_event();
+			let backing_event = Abi::backing_event();
+			let backing_address = EthereumBackingAddress::get();
+			let log_entry = verified_receipt
+				.logs
+				.into_iter()
+				.find(|x| {
+					x.address == backing_address &&
+						( x.topics[0] == register_event.signature()
+						  || x.topics[0] == backing_event.signature() )
+				})
+			.ok_or(<Error<T>>::LogEntryNE)?;
+
+			let input = if log_entry.topics[0] == register_event.signature() {
+				let ethlog = Self::parse_event(register_event, log_entry)?;
+				Self::process_erc20_creation(backing_address, ethlog)?
+			} else {
+				let ethlog = Self::parse_event(backing_event, log_entry)?;
+				Self::process_token_issuing(ethlog)?
+			};
+
+			let contract = MappingFactoryAddress::get();
+			let account = Self::dvm_account_id();
+			let basic = <T as darwinia_evm::Config>::AccountBasicMapping::account_basic(&account);
+			let substrate_account = <T as darwinia_evm::Config>::AddressMapping::into_account_id(account.clone());
+
+			<T as Config>::RingCurrency::transfer(&user, &substrate_account, T::FeeEstimate::get(), KeepAlive)?;
+			let transaction = Self::unsigned_transaction(basic.nonce, contract, input);
+			let result = dvm_ethereum::Module::<T>::do_transact(transaction, Some(account)).map_err(|e| -> &'static str {
+				log::trace!(target: "darwinia-issuing", "register_or_issuing_erc20 error {:?}", &e);
+				e.into()
+			} )?;
+
+			let leaved_balance = <T as Config>::RingCurrency::free_balance(&substrate_account);
+			// we should reserve some balance to keepalive this system account
```

Comment:
Abnormal,

Question: what will happen to substrate_account if the substrate account data is deleted? it's evm account data will always not be deleted right?